src/main/java/org/apache/sling/xss/impl/xml/Policy.java

```diff
@@ -342,19 +340,17 @@ private List<Attribute> getTagAttributes(List<Attribute> attributeList, String t
                 String description = attribute.getDescription();
 
                 // attribute has no children
-                if (regexps == null && literals == null) {
+                if (regexps.isEmpty() && literals.isEmpty()) {
```

Comment:
@nonanalou Here was the core problem of the failing tests. Once you return non-null lists from `Attribute` this check failed and common-attribute definitions were never used.